packages/arb-node-core/staker/staker_test.go

```diff
@@ -192,11 +212,36 @@ func TestStaker(t *testing.T) {
 		t.Fatal("Staker isn't staked")
 	}
 
-	if stakerInfo.CurrentChallenge != nil || stakerInfo.AmountStaked.Cmp(big.NewInt(0)) == 0 {
-		t.Fatal("Staker didn't resolve challenge")
+	if stakerInfo.CurrentChallenge != nil {
+		t.Fatal("Staker remained in challenge")
 	}
 
 	if stakerInfo.LatestStakedNode.Cmp(big.NewInt(0)) == 0 {
 		t.Fatal("Staker didn't stake on node")
 	}
+
+	faultyStakerInfo, err := staker.rollup.StakerInfo(ctx, common.NewAddressFromEth(validatorAddress2))
+	test.FailIfError(t, err)
+
+	if faultsExist {
+		if faultyStakerInfo != nil {
+			t.Fatal("Faulty staker is still staked")
+		}
+	} else {
+		if faultyStakerInfo == nil {
+			t.Fatal("Other staker lost stake")
+		}
+	}
+}
+
+func TestChallengeRegular(t *testing.T) {
+	adversarialTest(t, challenge.FaultConfig{DistortMachineAtGas: big.NewInt(10000)}, big.NewInt(100000000))
+}
+
+func TestChallengeTimeout(t *testing.T) {
+	adversarialTest(t, challenge.FaultConfig{DistortMachineAtGas: big.NewInt(50000)}, big.NewInt(100000))
+}
```

Comment:
Can we actually verify in the test how the challenge ended? Easiest way would probably be via checking to see if the expected was emitted